v2/pkg/engine/plan/visitor.go

```diff
@@ -383,6 +383,16 @@ func (v *Visitor) resolveFieldInfo(ref, typeRef int, onTypeNames [][]byte) *reso
 		parentTypeNames = append(parentTypeNames, onTypeName)
 	}
 
+	if v.Walker.EnclosingTypeDefinition.Kind == ast.NodeKindInterfaceTypeDefinition {
```

Comment:
What if the enclosing type is a union?

Reply:
Havent tested, will test

Reply:
DIscussed with david, will create a new ticket and handle it